src/modules/navigation/duck/hooks.ts

```diff
@@ -1,33 +1,89 @@
-import { useCallback } from 'react'
+import { useCallback, useRef } from 'react'
 import { useDispatch, useSelector } from 'react-redux'
 
-import flag from 'cozy-flags'
+import { useClient, Q, useQuery } from 'cozy-client'
 
 import { DEFAULT_SORT, SORT_BY_UPDATE_DATE } from '@/config/sort'
 import { TRASH_DIR_ID } from '@/constants/config'
+import { DOCTYPE_DRIVE_SETTINGS } from '@/lib/doctypes'
+import logger from '@/lib/logger'
 import { sortFolder, getSort } from '@/modules/navigation/duck'
 
+export interface Sort {
+  attribute: string
+  order: string
+}
+
+interface QueryResult {
+  data: [
+    | {
+        attributes: Sort
+      }
+    | undefined
+  ]
+  fetchStatus?: string
+}
+
 const useFolderSort = (folderId: string): [Sort, (props: Sort) => void] => {
+  const client = useClient()
+
   const defaultSort: Sort =
-    flag('drive.default-updated-at-sort.enabled') || folderId === TRASH_DIR_ID
+    folderId === TRASH_DIR_ID || folderId === 'recent'
       ? SORT_BY_UPDATE_DATE
       : DEFAULT_SORT
+
   const dispatch = useDispatch()
-  // eslint-disable-next-line @typescript-eslint/no-unsafe-return
-  const currentSort = (useSelector(state => getSort(state)) ||
-    defaultSort) as Sort
-  const setOrder = useCallback(
-    ({ attribute, order }: Sort) => {
+
+  const driveSettingsResult = useQuery(Q(DOCTYPE_DRIVE_SETTINGS), {
+    as: DOCTYPE_DRIVE_SETTINGS
+  }) as QueryResult
+
+  const initRef = useRef(false)
+  if (!initRef.current && driveSettingsResult.fetchStatus === 'loaded') {
+    const settings = driveSettingsResult.data[0]?.attributes
+    const currentSortOrder = settings?.order ?? defaultSort.order
+    const currentSortAttribute = settings?.attribute ?? defaultSort.attribute
+    dispatch(sortFolder(folderId, currentSortAttribute, currentSortOrder))
+    initRef.current = true
+  }
```

Comment:
please keep it simple 

```suggestion
const [settingsInit, setsettingsInit] = useState(false)

if (!settingsInit && driveSettingsResult.fetchStatus === 'loaded') {
  const settings = driveSettingsResult.data[0]?.attributes

  if (settings) {
    dispatch(sortFolder(folderId, settings.attribute, settings.order))
  }
  
  setsettingsInit(true)
}
```